apps/opencs/view/render/terrainshapemode.cpp

```diff
@@ -1038,14 +1038,29 @@ bool CSVRender::TerrainShapeMode::isInCellSelection(int globalSelectionX, int gl
     return false;
 }
 
+void CSVRender::TerrainShapeMode::handleSelection(int globalSelectionX, int globalSelectionY, std::vector<std::pair<int, int>>* selections)
+{
+    if (isInCellSelection(globalSelectionX, globalSelectionY)) selections->emplace_back(globalSelectionX, globalSelectionY);
+    else
+    {
+        int moduloX = globalSelectionX % (ESM::Land::LAND_SIZE - 1);
+        int moduloY = globalSelectionY % (ESM::Land::LAND_SIZE - 1);
+        bool xIsAtCellBorder = moduloX == 0;
+        bool yIsAtCellBorder = moduloY == 0;
+        if (isInCellSelection(globalSelectionX - 1, globalSelectionY) && xIsAtCellBorder && !yIsAtCellBorder) selections->emplace_back(globalSelectionX, globalSelectionY);
+        if (isInCellSelection(globalSelectionX, globalSelectionY - 1) && !xIsAtCellBorder && yIsAtCellBorder) selections->emplace_back(globalSelectionX, globalSelectionY);
+        if (isInCellSelection(globalSelectionX - 1, globalSelectionY - 1) && xIsAtCellBorder && yIsAtCellBorder) selections->emplace_back(globalSelectionX, globalSelectionY);
```

Comment:
isInCellSelection is not a trivial check, yet it's done thrice for each vertex of the selection the first isInCellSelection fails for unnecessarily. Maybe this part can be rearranged as
```cpp
if (!xIsAtCellBorder && !yIsAtCellBorder)
    return;
int selectionX = globalSelectionX;
int selectionY = globalSelectionY;
if (xIsAtCellBorder)
    selectionX--;
if (yIsAtCellBorder)
    selectionY--;
if (isInCellSelection(selectionX, selectionY))
    selections->emplace_back(globalSelectionX, globalSelectionY);
```

Please avoid making lines that long, we're not at shortage of line breaks.